sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -685,15 +711,22 @@ public DigitalTwinsServiceVersion getServiceVersion() {
             .map(
                 objectPagedResponse -> {
                     List<T> list = objectPagedResponse.getValue().stream()
-                        .map(object -> mapper.convertValue(object, clazz))
+                        .map(object -> {
+                            try {
+                                return DeserializationHelpers.castObject(mapper, object, clazz);
+                            } catch (JsonProcessingException e) {
+                                logger.error("JsonProcessingException occurred while retrieving relationships: ", e);
```

Comment:
here as elsewhere as well - the exception is during the deserialization, so maybe we should highlight that (it didn't occur during the REST operation).